test/e2e/config/azure-dev.yaml

```diff
@@ -149,6 +149,7 @@ providers:
 
 variables:
   AKS_KUBERNETES_VERSION: "latest"
+  AKS_KUBERNETES_MAX_VERSION: "1.23"
```

Comment:
I might be missing something but any particular reason why we're not just setting `AKS_KUBERNETES_VERSION` to `stable-1.23` to achieve the same result?